AttitudeManager/ControlAlgorithms/Src/AM_FixedControl.cpp

```diff
@@ -1,17 +1,116 @@
+    float targetAltitude = instructions.z_dir; 
+    float targetPitch = instructions.x_dir * maxPitchAngle;
+    float targetBank = instructions.y_dir * maxBankAngle;
+     
+    // Adjust heading difference for the PID
+    if (targetHeading <= (-180)) {
+        targetHeading += 180;
+    } else if (targetHeading > 180){
+        targetHeading -=360;
+    }
+   
+    // Define PID values for mixing
+    PIDController pid_bank{bank_kp,        bank_ki,      bank_kd, 
+                           bank_i_windup, -maxBankAngle, maxBankAngle };
+    PIDController pid_rudder{rudder_kp,       rudder_ki, rudder_kd, 
+                             rudder_i_windup, -100,      100};
+    PIDController pid_pitch{pitch_kp,       pitch_ki,       pitch_kd, 
+                            pitch_i_windup, -maxPitchAngle, maxPitchAngle};
+    PIDController pid_airspeed{airspeed_kp,       airspeed_ki,  airspeed_kd, 
+                               airspeed_i_windup, airspeed_min, airspeed_max};                        
+
+    // Compute values for mixing
+    float bank = pid_bank.execute(targetBank, currentAttitude.roll, currentAttitude.rollRate);
+
+    float pitch = pid_pitch.execute(targetPitch, currentAttitude.pitch, currentAttitude.pitchRate);
+
+    float rudderSet = rudderPercent(bank);
+    
+    rudderSet = pid_rudder.execute(0.0f, targetHeading);
 
-    // do something
+    float airspeed = pid_airspeed.execute(targetAltitude, currentAttitude.altitude, // handling altitude with airspeed? Define altitude and airspeed seperately?
+                                          currentAttitude.rateOfClimb);
+    
+    float roll = DEG_TO_RAD(bank);
+
+    // mix the PID's
+    float engineOutput = 
+        mixPIDs(configs[Engine].stateMix, roll, pitch, rudderSet, airspeed); 
+    float leftAileronOutput = 
+        mixPIDs(configs[LeftAileron].stateMix, roll, pitch, rudderSet, airspeed);
+    float rightAileronOutput = 
+        mixPIDs(configs[RightAileron].stateMix, roll, pitch, rudderSet, airspeed);
+    float rudder = 
+        mixPIDs(configs[Rudder].stateMix, roll, pitch, rudderSet, airspeed);
+    float elevator = 
+        mixPIDs(configs[Elevator].stateMix, roll, pitch, rudderSet, airspeed);
+
+    // Verify values
+    assert(configs[Engine].channel < outputsLength); // Question: Engine controlled as some percentage? Indicative of airspeed?
+    outputs[configs[Engine].channel] = engineOutput;
+
+    assert(configs[LeftAileron].channel < outputsLength); 
+    outputs[configs[LeftAileron].channel] = leftAileronOutput;
+        
+    assert(configs[RightAileron].channel < outputsLength);
+    outputs[configs[RightAileron].channel] = rightAileronOutput;
+        
+    assert(configs[Rudder].channel < outputsLength);
+    outputs[configs[Rudder].channel] = rudder;
+        
+    assert(configs[Elevator].channel < outputsLength);
+    outputs[configs[Elevator].channel] = elevator;
+
+}
+// Modify for fixed wing?
```

Comment:
I think your intuition is correct, try to think about the differences between a fixed wing and quad (I believe you copied aidans level quad it looks very similar).

Reply:
Regarding the use of assert, Aidan mentioned how you can get caught in a forever loop when your assert fails. I wondered what the intended outcome or state is if we fail an assertion. Apologies if this has been addressed.

Reply:
usually if an assert fails, we'll return an exit code and head to fatal failure or otherwise disarmed/safe state.

We don't actually have many asserts in our code right now, but we should start employing them. Misra C and barr C standards are good places to look about safety-critical applications of C and C++